packages/client/src/Dashboard.tsx

```diff
@@ -120,14 +121,23 @@ const useStyles = makeStyles(theme => ({
 export default function Dashboard({ accountId, navigate }: Props) {
   const classes = useStyles()
   const [open, setOpen] = React.useState(false)
-  const [isSearching, setIsSearching] = React.useState(false)
-  const [searchQuery, setSearchQuery] = React.useState("")
+  // const [isSearching, setIsSearching] = React.useState(false)
+
+  const [query, setQuery] = useQueryParams({
+    searchQuery: StringParam,
+    isSearching: BooleanParam
+  })
+  const { searchQuery, isSearching } = query
```

Comment:
Instead of checking for `undefined` later, you can assign default values in a destructuring assignment, like this:

```ts
const { searchQuery = "", isSearching = false } = query
```

An even more terse option which would avoid the intermediate `query` variable is nested destructuring:

```ts
const [{ searchQuery = "", isSearching = false }, setQuery] = useQueryParams({
  searchQuery: StringParam,
  isSearching: BooleanParam
})
```

I think the names `query` and `setQuery` are not descriptive. I would call them `searchParams` and `setSearchParams`.

But that still leaves an issue: `searchQuery` and `isSearching` are independent. Using a single hook for both pushes you into wanting to update both together. I think it would be better to use two hooks so that these values can be set separately as they were when they were backed by `useState`:

```ts
  const [searchQuery = "", setSearchQuery] = useQueryParam(
    "searchQuery",
    StringParam
  )
  const [isSearching = false, setIsSearching] = useQueryParam(
    "isSearching",
    BooleanParam
  )
```